File: sach/de3316.py
//DE3316.PY - Tran Huu Nam - hsgtin.vn - 2024
import sys
def main():
    n = int(input())
    D = [False] * 41
    dem = 0
    for _ in range(n):
        x = int(input())
        D[x] = True
    for i in range(1, 41):
        if D[i]:
            dem += 1
    print(dem)
    for i in range(1, 41):
        if D[i]:
            print(i, end=" ")
if __name__ == "__main__":
    main()

File: sach/de3316.cpp
//DE3316.CPP - Tran Huu Nam - hsgtin.vn - 2024
#include <iostream>
using namespace std;
int main() {
    int n, x, dem=0;  bool D[41]={};
    cin>>n;
    for (int i=0; i<n; i++) {
        cin>>x;
        D[x]=true;
    }
    for (int i=1; i<=40; i++)
        if (D[i]) dem++;
    cout<<dem<<endl;
    for (int i=1; i<=40; i++)
        if (D[i]) cout<<i<<" ";
}

